src/core/EventBus.js

```diff
@@ -75,17 +82,33 @@ function EventBus() {
         handlers[type][idx] = null;
     }
 
-    function trigger(type, payload) {
+    function trigger(type, payload = {}, streamId = undefined, mediaType = undefined) {
```

Comment:
We are currently working on https://github.com/Dash-Industry-Forum/dash.js/issues/3379 and would like to introduce an additional parameter in the trigger function as well. In order to align this and avoid a large constructor I suggest we make the third parameter and object {streamId,mediaType,additonalParameter}. That way we can enhance it even further later

Reply:
Why not enriching payload parameter with all these additional parameters, for example:

```
eventBus.trigger(Events.INIT_FRAGMENT_NEEDED, {
    streamId: streamInfo.id,
    mediaType: type,
    representationId: currentRepresentationInfo.id
});
```

Reply:
I prefer having a third object in the function for all internal attributes which are not supposed to be dispatched to the client/app

Reply:
OK I see
we may get heavier source code, but nevermind
So this could be a 3rd function parameter called 'filters'? Or this may contains other parameters that could have other utility than even filtering?

Reply:
Yes basically everything which we can use for filtering and decision making when triggering the event. We will adjust the PR/request accordingly: https://github.com/Dash-Industry-Forum/dash.js/issues/3379